pallets/nomination-pools/src/lib.rs

```diff
@@ -0,0 +1,3296 @@
+	/// Create the main, bonded account of a pool with the given id.
+	pub fn create_bonded_account(id: PoolId) -> T::AccountId {
+		T::PalletId::get().into_sub_account_truncating((AccountType::Bonded, id))
+	}
+
+	/// Create the reward account of a pool with the given id.
+	pub fn create_reward_account(id: PoolId) -> T::AccountId {
+		// NOTE: in order to have a distinction in the test account id type (u128), we put
+		// account_type first so it does not get truncated out.
+		T::PalletId::get().into_sub_account_truncating((AccountType::Reward, id))
+	}
+
+	/// Get the member with their associated bonded and reward pool.
+	fn get_member_with_pools(
+		who: &T::AccountId,
+	) -> Result<(PoolMember<T>, BondedPool<T>, RewardPool<T>), Error<T>> {
+		let member = PoolMembers::<T>::get(who).ok_or(Error::<T>::PoolMemberNotFound)?;
+		let bonded_pool =
+			BondedPool::<T>::get(member.pool_id).defensive_ok_or(DefensiveError::PoolNotFound)?;
+		let reward_pool =
+			RewardPools::<T>::get(member.pool_id).defensive_ok_or(DefensiveError::PoolNotFound)?;
+		Ok((member, bonded_pool, reward_pool))
+	}
+
+	/// Persist the member with their associated bonded and reward pool into storage, consuming
+	/// all of them.
+	fn put_member_with_pools(
+		member_account: &T::AccountId,
+		member: PoolMember<T>,
+		bonded_pool: BondedPool<T>,
+		reward_pool: RewardPool<T>,
+	) {
+		bonded_pool.put();
+		RewardPools::insert(member.pool_id, reward_pool);
+		PoolMembers::<T>::insert(member_account, member);
+	}
+
+	/// Calculate the equivalent point of `new_funds` in a pool with `current_balance` and
+	/// `current_points`.
+	fn balance_to_point(
+		current_balance: BalanceOf<T>,
+		current_points: BalanceOf<T>,
+		new_funds: BalanceOf<T>,
+	) -> BalanceOf<T> {
+		let u256 = T::BalanceToU256::convert;
+		let balance = T::U256ToBalance::convert;
+		match (current_balance.is_zero(), current_points.is_zero()) {
+			(_, true) => new_funds.saturating_mul(POINTS_TO_BALANCE_INIT_RATIO.into()),
+			(true, false) => {
+				// The pool was totally slashed.
+				// This is the equivalent of `(current_points / 1) * new_funds`.
+				new_funds.saturating_mul(current_points)
+			},
+			(false, false) => {
+				// Equivalent to (current_points / current_balance) * new_funds
+				balance(
+					u256(current_points)
+						.saturating_mul(u256(new_funds))
+						// We check for zero above
+						.div(u256(current_balance)),
+				)
+			},
+		}
+	}
+
+	/// Calculate the equivalent balance of `points` in a pool with `current_balance` and
+	/// `current_points`.
+	fn point_to_balance(
+		current_balance: BalanceOf<T>,
+		current_points: BalanceOf<T>,
+		points: BalanceOf<T>,
+	) -> BalanceOf<T> {
+		let u256 = T::BalanceToU256::convert;
+		let balance = T::U256ToBalance::convert;
+		if current_balance.is_zero() || current_points.is_zero() || points.is_zero() {
+			// There is nothing to unbond
+			return Zero::zero()
+		}
+
+		// Equivalent of (current_balance / current_points) * points
+		balance(
```

Comment:
Here is the modified code, before it was like this : 
https://github.com/paritytech/substrate/blob/master/frame/nomination-pools/src/lib.rs#L2799

And in polkadot sdk it was changed to this : 
https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/nomination-pools/src/lib.rs#L2928